app/assets/javascripts/discourse/app/lib/sidebar/user/categories-section/category-section-link.js

```diff
@@ -79,6 +80,8 @@ export default class CategorySectionLink {
   }
 
   get route() {
-    return "discovery.category";
+    return this.currentUser?.sidebarTopicDestination === "unread"
+      ? "discovery.newCategory"
+      : "discovery.latestCategory";
```

Comment:
Same as https://github.com/discourse/discourse/pull/18594/files#r995337308. If the user has the magic config enabled:

1. Route to unread if category has unread topic count
2. Route to new if category has new topic count
3.  Route to latest